### PR TITLE
[MB-8084] Services Counselor sees a success/fail message after editing move contact info

### PR DESCRIPTION
**Pull request by hannaliebl**
## Description

GIVEN I am a services counselor

When I have made changes to service member contact info

Then I’m greeted with a success message if the update was successful and an error message if it was not

* Add `onUpdate` prop to `<CustomerInfo />` that is called with either `'success'` or `'error'` string
* Add `customerEditAlert` state to `<ServicesCounselingMoveInfo />` that updates this state on the update callback. It also listens to route changes so the alert is cleared on page navigation. I figured just moving this state up to the parent made the most sense. State can be passed via the router, but it's not easily cleared. I didn't think adding a separate context made sense, either, as this state is only being passed down/up one level.
* Add `customerEditAlert` prop to `<ServicesCounselingMoveDetails /> to display the alert

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make server_run
make client_run
```
Save move customer info to view the success alert.

Return a rejected promise from `updateCustomerInfo` in `ghcApi.js` to test the error state.

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-8084) for this change

## Screenshots

Screen Shot 2021-05-26 at 3 21 05 PM
Screen Shot 2021-05-26 at 3 22 14 PM

### CODE REVIEW

**Comment by hannaliebl** [comment]
Hmm, I merged master in and my tests are failing, I think because of this being added:
```
  const handleClose = () => {
    history.push(generatePath(servicesCounselingRoutes.MOVE_VIEW_PATH, { moveCode }));
  };
```
Which now depends on `moveCode` being defined, but it's not in the test. Is there something with the `MockProviders` setup we are doing wrong? Tagging @felipe-lee because you might have familiarity with this.

**Comment by felipe-lee** [comment]
> Hmm, I merged master in and my tests are failing, I think because of this being added:
> 
> ```
>   const handleClose = () => {
>     history.push(generatePath(servicesCounselingRoutes.MOVE_VIEW_PATH, { moveCode }));
>   };
> ```
> 
> Which now depends on `moveCode` being defined, but it's not in the test. Is there something with the `MockProviders` setup we are doing wrong, something that doesn't make `useParams` work? Tagging @felipe-lee because you might have familiarity with this.

I _think_ it might be because of the way the `initialEntries` are being defined currently. In another test file we define the url based on how it would be generated, which i guess sets it up to be pulled out in params as expected:
https://github.com/transcom/mymove/blob/35021ee2c8ae92aa9c071130141997f0efe33079/src/pages/Office/ServicesCounselingMoveDetails/ServicesCounselingMoveDetails.test.jsx#L179-L187

File: src/pages/Office/CustomerInfo/CustomerInfo.test.jsx [hunk 1]
```diff
@@ -53,4 +60,48 @@ describe('CustomerInfo', () => {
     expect(screen.getByLabelText('ZIP').value).toEqual(mockCustomer.current_address.postal_code);
     expect(screen.getByLabelText('Name').value).toEqual(mockCustomer.backup_contact.name);
   });
+
+  it('calls onUpdate prop with success on successful form submission', async () => {
+    const mockUpdate = jest.fn();
+    updateCustomerInfo.mockImplementation(() => Promise.resolve({ customer: { customerId: '123' } }));
+    render(
+      <MockProviders initialEntries={['moves/CDG3TR/customer']}>
```
Comment (felipe-lee) [inline src/pages/Office/CustomerInfo/CustomerInfo.test.jsx:68]: So here is where I would change what I said in https://github.com/transcom/mymove/pull/6699#issuecomment-849114089

File: src/pages/Office/CustomerInfo/CustomerInfo.test.jsx [hunk 2]
```diff
@@ -53,4 +60,48 @@ describe('CustomerInfo', () => {
...
+    await waitFor(() => {
+      expect(mockUpdate).toHaveBeenCalledWith('success');
+    });
+  });
+
+  it('calls onUpdate prop with error on unsuccessful form submission', async () => {
+    const mockUpdate = jest.fn();
+    updateCustomerInfo.mockImplementation(() => Promise.reject());
+    render(
+      <MockProviders initialEntries={['moves/CDG3TR/customer']}>
```
Comment (felipe-lee) [inline src/pages/Office/CustomerInfo/CustomerInfo.test.jsx:90]: And this one too: https://github.com/transcom/mymove/pull/6699#issuecomment-849114089

**Comment by hannaliebl** [comment]
@felipe-lee, ah, I should have mentioned, I did try to set the test up that way locally but it still failed. I will keep looking at this today.

**Review by amanda5iu — APPROVED** [review]

**Comment by felipe-lee** [comment]
> @felipe-lee, ah, I should have mentioned, I did try to set the test up that way locally but it still failed. I will keep looking at this today.

Sorry I didn't get back to this sooner. Just figured it out, you need something like this:
https://github.com/transcom/mymove/blob/35021ee2c8ae92aa9c071130141997f0efe33079/src/pages/Office/ServicesCounselingMoveDetails/ServicesCounselingMoveDetails.test.jsx#L18-L21

With the move code you want

**Review by Ryan-Koch — APPROVED** [review]

**Pull request merged**